### PR TITLE
#726 Ideal transformer implementation

### PR DESCRIPTION
**Pull request by adriguir**

### CODE REVIEW

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/IdealTransformerVariableTapI_INIT.mo [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+* This file is part of Dynawo, an hybrid C++/Modelica open source time domain simulation tool for power systems.
+*/
+
+model IdealTransformerVariableTapI_INIT "Initialization for ideal transformer based on the network voltage and current"
+
+  extends BaseClasses_INIT.BaseIdealTransformerVariableTap_INIT;
+  extends AdditionalIcons.Init;
+
+  protected
+    Types.ComplexVoltagePu u10Pu  "Start value of complex voltage at terminal 1 in p.u (base UNom)";
```
Comment (MarianneSaugierRTE) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/IdealTransformerVariableTapI_INIT.mo]: Should they be parameters if they are the inputs of this init model?

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/TransformerFixedRatio.mo [hunk 2]
```diff
@@ -12,7 +12,7 @@ within Dynawo.Electrical.Transformers;
 * This file is part of Dynawo, an hybrid C++/Modelica open source time domain simulation tool for power systems.
 */
 
-model TransformerFixedRatio "Two winding transformer with a fixed ratio"
+model TransformerFixedRatio "Two windings transformer with a fixed ratio"
```
Comment (MarianneSaugierRTE) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/TransformerFixedRatio.mo]: I always have doubts here: I think the right way to write this is Two-winding transformer, like a Three-year-old boy, can you maybe check with someone you trust on this?:)

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/BaseClasses_INIT.mo [hunk 3]
```diff
@@ -148,24 +180,78 @@ partial model BaseTransformerVariableTap_INIT "Base model for initialization of
...
+  Tap0 = IdealTransformerTapEstimation(rTfoMinPu, rTfoMaxPu, NbTap, u10Pu, Uc20Pu);
+
+  // Transformer equations
+  i10Pu = - rTfo0Pu * i20Pu;
+  rTfo0Pu * u10Pu = u20Pu;
+
+annotation(preferredView = "text");
+end BaseIdealTransformerVariableTap_INIT;
+
+partial model BaseNormalTransformerVariableTap_INIT "Base model for initialization of TransformerVariableTap"
```
Comment (MarianneSaugierRTE) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/BaseClasses_INIT.mo]: Remove "normal" from name?

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/TransformerFixedRatio.mo (hunk 2 above)
Comment (adriguir) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/TransformerFixedRatio.mo]: No you are right, it is two-winding transformer. My mistake ;)

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/BaseClasses_INIT.mo (hunk 3 above)
Comment (adriguir) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/BaseClasses_INIT.mo]: I wanted to keep similar names for the common part between ideal and classical transformers at initialization and time-domain stages. 
Thus I removed the normal here but added a Common in the base model name for the time-domain part so I have a symmetry between names in init and time-domain parts

Comment (MarianneSaugierRTE) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/BaseClasses_INIT.mo]: I don't see conflicts between base models so I would have expected the base model to be BaseTransformerVariableTap. You mention a symmetry with init models. If it is important to underline the symmetry I would have the "common" placed at the same place: BaseCommonBlabla or BaseBlaBlaCommon for both. But as their are all partial and won't be used as they are I don't think the symmetry is mandatory. So I let you decide between BaseTransformerVariableTap or BaseCommonTransformerVariableTap and BaseCommonTransformerVariableTap_INIT. What do you think?

Comment (adriguir) [reply dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/BaseClasses_INIT.mo]: I don't really know. This is not big deal in any case: thus I came back to the version without symmetry and without the Normal part in the name.

**Pull request merged**